commands/command_pre_push.go

```diff
@@ -83,8 +100,34 @@ func prePushCommand(cmd *cobra.Command, args []string) {
 			Print("Error scanning for Git LFS files in %q", left)
 			ExitWithError(err)
 		}
+
+		for _, p := range pointers {
+			if l, ok := lockSet[p.Name]; ok {
+				if l.Name == name && l.Email == email {
+					myLocks = append(myLocks, l.Path)
+				} else {
+					lockConflicts = append(lockConflicts, p.Name)
+				}
+			}
+		}
+
+		if len(lockConflicts) > 0 {
+			Error("Some files are locked in %s...%s", left, cfg.CurrentRemote)
+			for _, file := range lockConflicts {
+				Error("* %s", file)
+			}
+			os.Exit(1)
+		}
+
 		uploadPointers(ctx, pointers)
 	}
+
+	if len(myLocks) > 0 {
+		Print("Pushing your locked files:")
```

Comment:
I think it would be good to explain more what this means in practice - i.e. that the user should consider unlocking the files if they have no further changes to publish after this push.